storm-core/src/jvm/org/apache/storm/daemon/supervisor/BasicContainer.java

```diff
@@ -0,0 +1,644 @@
+                + " ";
+
+        List<String> command = mkProfileCommand(profileAction, stop, workerPid, targetDir);
+
+        File targetFile = new File(targetDir);
+        if (command.size() > 0) {
+            return runProfilingCommand(command, env, logPrefix, targetFile);
+        }
+        LOG.warn("PROFILING REQUEST NOT SUPPORTED {} IGNORED...", request);
+        return true;
+    }
+
+    /**
+     * Get the command to run when doing profiling
+     * @param action the profiling action to perform
+     * @param stop if this is meant to stop the profiling or start it
+     * @param workerPid the PID of the process to profile
+     * @param targetDir the current working directory of the worker process
+     * @return the command to run for profiling.
+     */
+    private List<String> mkProfileCommand(ProfileAction action, boolean stop, String workerPid, String targetDir) {
+        switch(action) {
+            case JMAP_DUMP:
+                return jmapDumpCmd(workerPid, targetDir);
+            case JSTACK_DUMP:
+                return jstackDumpCmd(workerPid, targetDir);
+            case JPROFILE_DUMP:
+                return jprofileDump(workerPid, targetDir);
+            case JVM_RESTART:
+                return jprofileJvmRestart(workerPid);
+            case JPROFILE_STOP:
+                if (stop) {
+                    return jprofileStop(workerPid, targetDir);
+                }
+                return jprofileStart(workerPid);
+            default:
+                return Lists.newArrayList();
+        }
+    }
+
+    private List<String> jmapDumpCmd(String pid, String targetDir) {
+        return Lists.newArrayList(_profileCmd, pid, "jmap", targetDir);
+    }
+
+    private List<String> jstackDumpCmd(String pid, String targetDir) {
+        return Lists.newArrayList(_profileCmd, pid, "jstack", targetDir);
+    }
+
+    private List<String> jprofileStart(String pid) {
+        return Lists.newArrayList(_profileCmd, pid, "start");
+    }
+
+    private List<String> jprofileStop(String pid, String targetDir) {
+        return Lists.newArrayList(_profileCmd, pid, "stop", targetDir);
+    }
+
+    private List<String> jprofileDump(String pid, String targetDir) {
+        return Lists.newArrayList(_profileCmd, pid, "dump", targetDir);
+    }
+
+    private List<String> jprofileJvmRestart(String pid) {
+        return Lists.newArrayList(_profileCmd, pid, "kill");
+    }
+
+    /**
+     * Compute the java.library.path that should be used for the worker.
+     * This helps it to load JNI libraries that are packaged in the uber jar.
+     * @param stormRoot the root directory of the worker process
+     * @param conf the config for the supervisor.
+     * @return the java.library.path/LD_LIBRARY_PATH to use so native libraries load correctly.
+     */
+    protected String jlp(String stormRoot, Map<String, Object> conf) {
```

Comment:
Nitpick: Consider not abbreviating this. It's obvious what it is here, but probably less so when called from elsewhere in the code.